### DISCUSSION TITLE
streaming nnvad

### DISCUSSION THREAD

**Discussion by YLQY** [pyannote/pyannote-audio#997] [Q&A]
Hi . I am following pyannote's vad model
The vad in pyannote is not good for streaming. I use 2s, or 1s or 0.5s chunks for the audio, and the result of doing vad is not very good. After reading SincNet, the first layer is self.wav_norm1d = nn.InstanceNorm1d(1, affine=True). The data is to use torchaudio to read audio, torchaudio will automatically normalize the entire audio, I found that the maximum and minimum results of normalization are not (-1, 1), nor (0, 1). Excuse me, if I want to train this model into a stream recognition model, can I not standardize the entire audio when reading the sample point data, but standardize the local chunk internally. Or want to change this model to flow recognition, do you have any good suggestions here?
thanks

**Comment by hbredin** [comment]
Here are a few links that might be relevant.

* [Streaming voice activity detection with pyannote.audio](https://herve.niderb.fr/fastpages/2021/08/05/Streaming-voice-activity-detection-with-pyannote.html)
* [Streaming speaker diarization based on pyannote.audio](https://github.com/juanmc2005/StreamingSpeakerDiarization)

**Comment by YLQY** [comment]
Thanks
